docs/product/components/modals.html

```diff
@@ -137,19 +186,31 @@ <h1 class="s-modal--header" id="modal-title">Example title</h1>
     <p class="stacks-copy">Not every modal is sunshine and rainbows. Sometimes there are potentially drastic things that could happen by hitting a confirm button in a modal—such as deleting an account. In moments like this, add the <code class="stacks-code">.has-danger</code> class to <code class="stacks-code">.s-modal</code>. Additionally, you should switch the buttons to <code class="stacks-code">.s-btn__danger.s-btn__filled</code>, since the main call to action will be destructive.</p>
     <div class="stacks-preview">
 {% highlight html linenos %}
-<aside class="s-modal has-danger js-modal-overlay js-modal-close" id="modal-base" tabindex="-1" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description" aria-hidden="true">
-    <div class="s-modal--dialog js-modal-dialog" role="document">
-        <h1 class="s-modal--header" id="modal-title">…</h1>
-        <p class="s-modal--body" id="modal-description">…</p>
-        <div class="grid gs8 gsx s-modal--footer">
-            <button class="grid--cell s-btn s-btn__filled s-btn__danger js-modal-primary-btn" type="button">…</button>
-            <button class="grid--cell s-btn s-btn__muted js-modal-cancel-btn" type="button">…</button>
+<div data-controller="s-modal">
+    <button data-action="s-modal#show">Launch example modal w/ danger state</button>
+    <aside
+        class="s-modal has-danger"
+        tabindex="-1" 
+        role="dialog"
+        aria-labelledby="modal-title"
+        aria-describedby="modal-description"
+        aria-hidden="true"
+        data-target="s-modal.content"
+        data-action="click->s-modal#hide"
+    >
+        <div class="s-modal--dialog" role="document" data-action="click->s-modal#nohide">
+            <h1 class="s-modal--header" id="modal-title">…</h1>
+            <p class="s-modal--body" id="modal-description">…</p>
+            <div class="grid gs8 gsx s-modal--footer">
+                <button class="grid--cell s-btn s-btn__filled s-btn__danger" type="button" data-action="s-modal#hide">…</button>
+                <button class="grid--cell s-btn s-btn__muted" type="button" data-action="s-modal#hide">…</button>
+            </div>
+            <a class="s-modal--close s-btn s-btn__muted" href="#" aria-label="Close" data-action="s-modal#hide">
+                @Svg.ClearSm
+            </a>
         </div>
-        <a class="s-modal--close s-btn s-btn__muted js-modal-close" href="#" aria-label="Close">
-            @Svg.ClearSm
-        </a>
-    </div>
-</aside>
+    </aside>
+</div>
 {% endhighlight %}
```

Comment:
I can handle this, but we'll want to add a table with all the data attributes explaining where to add them and what they do.